department-of-veterans-affairs/abd-vro
# hook up VRO gateway to python for domain-cc

This PR addresses the remaining 2 subtasks for [Figure out how to talk from CC to VRO gateway#41](https://github.com/department-of-veterans-affairs/vagov-claim-classification/issues/41).
This is a continuation from https://github.com/department-of-veterans-affairs/abd-vro/pull/1517

## What was the problem?
API calls from inside the Java app container of VRO need to trigger python code to run.
This is done by
-  sending a message to rabbit mq from a Java controller class
- having a python client receive messages
- having a python client host a python Web api
- when the python RabbitMQ client receives messages, it invokes the corresponding python endpoint

Associated tickets or Slack threads:

- [Figure out how to talk from CC to VRO gateway#41](https://github.com/department-of-veterans-affairs/vagov-claim-classification/issues/41)

## How does this fix it?[^1]

- Python script added for communicating between rabbitMQ and FastAPI
- Java Controller class updated to forward the payload and status code based on contents of RabbitMQ message

## How to test this PR
- see "testing it all together" in [abd-vro/domain-cc/README.md](https://github.com/department-of-veterans-affairs/abd-vro/tree/feature-rabbit-avoid-lhapi/domain-cc#testing-it-all-together)


[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).
